### ISSUE TITLE
FSPagerView inside UITableView : need to set pagerView currentIndex before it gets displayed

### ISSUE THREAD

**Issue by lucasgerard** [WenchaoD/FSPagerView#54]
Hi,

I'm currently using `FSPagerView` inside a `UITableView`, just as it was a `CollectionView` from this [article]( https://ashfurrow.com/blog/putting-a-uicollectionview-in-a-uitableviewcell-in-swift/)

However, I can't find a way to set the `contentOffset` of the `pagerView` as I would for a `collectionView` (`collectionView.contentOffset.x = newValue`), to ensure that when my `tableView` cells are reused, the offset are preserved and my `pagerView` is displayed with the right `currentIndex`.

Basically, I would like to set the pagerView current displayed index in my tableView `WillDisplayCell` method, but as I try using `scrollToItem` it crashes with an index `out of range` issue showing that `numberOfItems` is still 0, as its data was not loaded yet.

Any idea how to accomplish that? Hope I'm clear enough, thanks for your help!

**Comment by WenchaoD** [comment]
The content offset is useless if your collectionView hasn't layout it's content. Typically you should set the content offset in `viewDidLayoutSubviews` function of your view controller, or force the collection view layout before scrolling by calling `layoutIfNeeded()`. In this case, the following code could be helpful.
```swift
pagerView.layoutIfNeeded()
pagerView.scrollToItem(at: index, animated: false)
```

**Comment by lucasgerard** [comment]
Thanks for the help, it worked!

**Comment by WenchaoD** [comment]
You are welcome.
